### PR TITLE
wrong state and country for packages hc and fhc

### PR DESCRIPTION
**Pull request by j-ibrahim**
1) Changes for correcting state name and country name for HC and FHC - Search and Infosite
2) stateProvinceCode correction for all package combination - search and infosite

### CODE REVIEW

File: src/library/simplified/extension/page-name-javascript-functions.js [hunk 1]
```diff
@@ -358,6 +358,13 @@ window.utag.isPIS_HC = function() {
     return false;
 }
 
+window.utag.isPIS_FHC = function(){
+    if(pageName.indexOf("page.FHC.Package.Infosite.Information") > -1){
```
Comment (wmoussa) [inline src/library/simplified/extension/page-name-javascript-functions.js]: you can just return the boolean expression

`return pageName.indexOf("page.FHC.Package.Infosite.Information") > -1`

File: src/library/simplified/extension/country-code.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ if((utag.isFRateDetails() || utag.isPRateDetails())  && typeof b["entity.tripDet
 else if(utag.isFCO() && typeof b["entity.checkout.flightOffer.arrivalCountryName"] != "undefined"){
     b["countryCode"] = b["entity.checkout.flightOffer.arrivalCountryName"];
 }
-else if(utag.isCarSR()){
+else if((utag.isCarSR() || utag.isPCarSearch()) && b["entity.carSearch.searchCriteria.dropOffRegion.countryCode"] != undefined){
```
Comment (wmoussa) [inline src/library/simplified/extension/country-code.js:7]: shouldn't you use typeof? like above?

Comment (wmoussa) [reply src/library/simplified/extension/country-code.js:7]: if it is not needed, ignore my comment

File: src/library/simplified/extension/state-name.js [hunk 3]
```diff
@@ -29,14 +29,7 @@ else if(utag_data["pageInfo.pageName"] == "page.Package-Search.Mobile,P,20" && u
...
 }
-else if((utag.isPSR_HC() || utag.isPIS_HC() || utag.isPSR_FHC())
-    && (typeof b.entity.packageSearch.packageSearchParameters != "undefined" && typeof b.entity.packageSearch.packageSearchParameters.flightSearchParameters != "undefined")){
-    var stateSplit = b.entity.packageSearch.packageSearchParameters.flightSearchParameters.departureAirportCityState.split(",").length;
-    if(stateSplit >= 2){
-        utag_data["state"] = b.entity.packageSearch.packageSearchParameters.flightSearchParameters.departureAirportCityState.split(",")[stateSplit - 2];
-    }
-}
-else if((utag.isPSR_FH_Responsive() || utag.isPIS_FH() || utag.isPSR_FC()) 
+else if((utag.isPSR_FH_Responsive() || utag.isPIS_FH() || utag.isPSR_FC() || utag.isPSR_FHC() || utag.isPIS_FHC() || utag.isPSR_HC() || utag.isPIS_HotelCar())
```
Comment (wmoussa) [inline src/library/simplified/extension/state-name.js]: could we group those boolean checks in a single method? How would you name it?

File: src/library/simplified/extension/state-province-code.js [hunk 4]
```diff
@@ -32,9 +32,9 @@ else if(utag.isPSR() && b['entity.packageSearch.packageSearchParameters.flightSe
         b["stateProvinceCode"] = b["stateProvinceCode"].split("(")[0].trim();
     }
 }
-else if(utag.isPSR_FHC() && b["entity.packageSearch.packageSearchParameters.flightSearchParameters.departureAirportCityState"] != undefined)
+else if((utag.isPSR_FH_Responsive() || utag.isPIS_FH() || utag.isPSR_FC() || utag.isPSR_FHC() || utag.isPIS_FHC() || utag.isPSR_HC() || utag.isPIS_HotelCar()) && b["entity.packageSearch.packageSearchParameters.flightSearchParameters.arrivalAirportCityState"] != undefined)
```
Comment (wmoussa) [inline src/library/simplified/extension/state-province-code.js]: this if statement is also hard to follow, please encapsulate in a method and name it accordingly.

File: src/library/simplified/extension/country-code.js (hunk 2 above)
Comment (j-ibrahim) [reply src/library/simplified/extension/country-code.js:7]: both works
